### ISSUE TITLE
Fresh install of Wrapper.py in certain conditions does this thing

### ISSUE THREAD

**Issue by benbaptist** [benbaptist/minecraft-wrapper#262]
```
[19:57:15] [Wrapper.py/ERROR] Traceback (most recent call last):
[19:57:15] [Wrapper.py/ERROR]   File "Wrapper.py/server.py", line 219, in reloadProperties
[19:57:15] [Wrapper.py/ERROR]     self.worldName = self.properties.get("main", "level-name")
[19:57:15] [Wrapper.py/ERROR]   File "/usr/lib/python2.7/ConfigParser.py", line 618, in get
[19:57:15] [Wrapper.py/ERROR]     raise NoOptionError(option, section)
[19:57:15] [Wrapper.py/ERROR] NoOptionError: No option 'level-name' in section: 'main'
```

**Comment by suresttexas00** [comment]
interesting error... never seen this one before.  Sounds like it can't access server properties or there is no level-name item... If so, I can think one one instance where level-name would not exist: A fresh server install where the EULA is set to false.

**Comment by benbaptist** [comment]
Yup! And that's precisely where it was happening! I need to have more error checks - even once the server fully booted (after changing eula.txt), I think it was still working off an old server.properties or something so it was a bit screwy until I rebooted Wrapper.py itself.

**Comment by suresttexas00** [comment]
That is such a rare thing to happen.  I believe the best way to fix that is just do a 

```
try:
    self.eula = self.properties.get("main", "eula")  # not sure what this returns?  Boolean True or string "true"?
except
    self.eula = "false"  # presuming string??
if self.eula == "false": 
    Print("Server cannot start.. EULA not found or not accepted")
    blah blah systemexit
```

**Comment by suresttexas00** [comment]
Here's something I am working on... making a separate module that will allow referencing the packet numbers by name.  Not only might this make it easier to do different versions, but it might even allow the clients and servers to be on different versions (even though the gameplay in that would be hazardous and buggy to a player due to different blocks and creatures, etc between versions).  What I am working on is just the packet numbering, but it could be expanded to include the parsing definitions for sending/receiving too.

```

# Ways to reference packets by names and not hard-coded numbers
# It is up to the plugin or wrapper to know what the actual version connection is.
# Once it knows the version, it can do something like:
#  `from mcpkt import serverBound18 as ClPkt`
#  the remainder of the wrapper/plugin code can simply reference `ClPkt.playerlook`


class ServerBound18:
		''' wrapper's "Client" process, which handles connections from client to wrapper.
		
		These packets are being sent to the server (i.e., wrapper's proxy) from the client.
		Proxy, in turn, can "send" these on, or drop them (return False)
		'''
			
	def __init__(self)	
		# lines 205ff
		# 1.8.8 packets
		self.keepalive = 0x00  # client's response to server challenge
		self.chatmessage = 0x01
		self.useentity = 0x02
		self.player = 0x03  # onground
		self.playerposition = 0x04
		self.playerpos_look = 0x06
		self.playerlook = 0x05
		self.playerblockdig = 0x07
		self.playerblockplace = 0x08
		self.helditemchange = 0x09
		self.playerupdatesign = 0x12
		self.clientsettings = 0x15
		self.spectate = 0x18
		self.useitempacket = "notExists"
		self.clickwindow = 0x0e
```

**Comment by suresttexas00** [comment]
I have currently coded these self definitions into the proxy version I am using to support both 1.8 and 1.9 snaps.

**Comment by suresttexas00** [comment]
The other reason I think I like this approach is that the plugins can send packets without needing to even know what the version of the game is!  Since Wrapper will have already imported the applicable set under a common name, the plugins just need to reference it using the wrapper naming.

**Comment by JasonBristol** [comment]
@suresttexas00  I think abstracting packets in this regard is a good idea, this should also be relatively simple to implement/expand if you already have a working version. Very interesting the prospect of cross version compatibility.

**Comment by suresttexas00** [comment]
Yes, I actually have a version similar already working for 15w40b on my production server.  The abstraction is done in proxy.py and I am just moving the packet definitions out of proxy.py with this update... and updating the packets for 15w51b.  In the process, I am cleaning it back up to more similar to Ben's version (restoring keep alives and such).

**Comment by suresttexas00** [comment]
I am also trying to remove some of the `elif` blending of gameplay and non-gameplay parsing.  Here, for instance 0x00 is parsed for two different states... which is not going to work for 1.9 because of the packet renumbering:
```
	def parse(self, id, original):
		if id == 0x00:
			if self.state < 3:
				message = self.read("string:string")
				self.log.info("Disconnected from server: %s" % message["string"])
				self.client.disconnect(message)
				return False
			elif self.state == 3:
				if self.client.version > 7:
					id = self.read("varint:i")["i"]
					if not id == None:
						self.send(0x00, "varint", (id,))
				return False
```

to..

```
	def parse(self, id, original):  # client - bound parse ("Server" class connection)
		"""
		Client-bound "Server Class"
		"""
		# disconnect, I suppose...
		if id == 0x00 and self.state < 3:
			message = self.read("string:string")
			self.log.info("Disconnected from server: %s" % message["string"])
			self.client.disconnect(message)
			return False
			
		# handle keep alive packets 
		if id == self.pktCB.keepalive and self.state == 3:
				if self.client.version > 7:
					id = self.read("varint:i")["i"]
					if not id == None:
						self.send(self.pktSB.keepalive, "varint", (id,))
				return False
``` 

At this point, I am just working on gameplay state packets because that is all the has changed thus far (thankfully!!)

**Comment by suresttexas00** [comment]
A lot more work will be needed to actually have total version independence (if even possible) since each part (Server/Client) references the the same "set" of classes (1.9 or 1.8).  So even if you told the client side. "I will use 1.8", any server.send activity will use the 1.8 class to send, crapping the 1.9 server out.

**Comment by Sasszem** [comment]
It happened to me, but after restart, It worked without errors(except that I caused).

**Comment by suresttexas00** [comment]
Yes, because it is simply caused by a missing file that will normally be restored after the server boots.  I had it happen yesterday because I intentionally deleted the level.dat file because my 16w04a server "End" level would not spawn an enderdragon (even if I deleted all the DIM1 files and /summon him).  Of course afterwards, I had to reset all my game elements like keepInventory and doFireTick, etc...

Ben, I don't think this is needed.  It would only happen (sometimes) on the initial startup of a server from scratch.
